### ISSUE TITLE
Add an option to application plugin to genererate a PowerShell script.

### ISSUE THREAD

**Issue by sydist** [gradle/gradle#21081]
**Environment (Windows 11, WSL 2)**

I've been trying to run a `gradlew.bat` script that was inside my WSL 2 _(Windows Subsystem for Linux)_ environment _(because that's where I do basically all my coding)_, but it's not possible because CMD doesn't support UNC paths so I can't even access the folder the script is in with cmd, however, powershell does, changing the .bat script to a .ps1 script should fix this problem.

The reason I'm trying to run the `gradlew.bat` for windows instead of the `gradlew` for linux, is because I'm running a task that boots up a game for testing, and it runs really slow on wsl, so I have to run the task with windows but keep the development environment in linux for optimum efficiency

**Comment by ciscoo** [comment]
> Replacement is not an option

@jbartok can you elaborate on why it is not an option rather than declining altogether.

**Comment by ov7a** [comment]
This is a valid feature request.

Note that the PS option should be configurable. This option should be propagated to the wrapper task.

I've changed the title of the issue to reflect that.

**Comment by jbartok** [comment]
> > Replacement is not an option
> 
> [@jbartok](https://github.com/jbartok) can you elaborate on why it is not an option rather than declining altogether.

Yes, I've been a bit blunt in my reply. Let me try to write a more complete one. We could (and we would support to) do the following:

1. add `.ps1` first, although at first, it would be an opt-in to not break existing builds that don't expect a ps1 to be generated
2. start deprecating `.bat`
3. remove `.bat` and make the `.ps1` the default

Leveraging Gradle 9.0 to affect this change would indeed very much make sense and would simplify the process, but unfortunately, we don't have the resources. Gradle 9.0 is already planned to be a huge effort, and more strategically important changes haven't left room for ones like this. 

It will need to happen later, even though it will be harder to achieve in minor versions.

**Comment by Vampire** [comment]
Please do **not** deprecate or remove BAT file but at most provide a PS1 as additional option or configurable.

PS1 is not always an option, for example
- for me personally using Powershell as shell is an absolute no-go, as it is unusable for me, for example the shell trying to be smarter than its user, for example if you execute `.\gradlew help -Dorg.gradle.debug=true`, it does some argument processing at the dot, resulting in the error `Task '.gradle.debug=true' not found in root project 'showcase'.` unless you quote the argument like `.\gradlew help "-Dorg.gradle.debug=true"` which is one of the most annoying things I have ever seen in a shell.
  From cmd when trying to invoke a PS1 file, by default only an editor with the content opens.
  From Git Bash when trying to invoke a PS1 file, it tries to execute it as Bash script.
- PS1 scripts are not necessarily executable, especially when marked as downloaded form somewhere.
  Many Windows instances by default forbid execution of PS1 scripts completely.
  And even if PS1 scripts are allowed at all, they could still be restricted to only run ones that are digitally signed by a valid bought code-signing certificate.
  And even if locally written PS1 scripts are allowed to be executed, it could still be configured that remotely downloaded PS1 scripts require the digital signature.
  For one of our projects where we started to use PS1 scripts for the startup and arguments logic, we had significant problems as they did not run out-of-the-box for several customers and even on colleague's boxes they refused to run when downloaded from CI, even when they were digitally signed.
  We resorted to have a `bat` file that the customers run, that then uses `powershell.exe -ExecutionPolicy RemoteSigned ...` to hand off to the signed PS1 scripts as this option overrides the system-level execution policy and only then those problems were gone so far.
  So to locally run a PS1 script from a powershell shell, you need to change your execution policy if you have the right to do so.
  To run it from a different shell, you always have to prefix it with `powershell -ExecutionPolicy RemoteSigned ` or similar which is annoying.

On the other hand starting a BAT file from powershell of course works flawlessly.

So to summarize:
PS1 Skripts are fine if you only use them for yourself and know your environment allows you to run them.
But generating them in something you want other people to use where the execution environment is not under your control is imho a very bad and user-unfriendly option unless you have a BAT file that cares about properly handing over to powershell.

So better only use PS1 if you write those scripts yourself and need the increased power you have in them over what BAT files provide, and you are aware of the drawbacks and problems they might produce.

For those comparatively simple scripts that Gradle generates that are fully functioning and don't need the increased possibilities a PS1 script would provide, I'd strongly ask to just stay with the BAT files.

**Comment by Vampire** [comment]
Btw. @sydist 

> but it's not possible because CMD doesn't support UNC paths so I can't even access the folder the script is in with cmd,

I cannot follow, cmd can access UNC paths just fine on Windows 11:

<img width="585" height="479" alt="Image" src="https://github.com/user-attachments/assets/3f9c1019-32b4-4f14-bff0-78f32e123ea3" />

> however, powershell does, changing the .bat script to a .ps1 script should fix this problem.

And even if it would not, you can anytime just call a BAT file from powershell just fine:

<img width="924" height="317" alt="Image" src="https://github.com/user-attachments/assets/c9c62d20-eacc-4115-8b20-d26bd1f2e5c0" />

> The reason [...] it runs really slow on wsl, so I have to run the task with windows but keep the development environment in linux for optimum efficiency

Make sure your files are not on a Windows drive, but on a WSL drive.
With WSLv1 the file access was much faster because that was more an API emulation layer and file access was fast.
With WSLv2 the *nix distribution is a completely separate VM which makes the *nix more native and everything just working much better, but the file accesses from WSLv2 distribution to the Windows filesystem (and the other way around) is awfully slow. So if you make sure that your build is only accessing files within the WSLv2 filesystem, the build should usually be comparatively fast to having the files on Windows filesystem and building / running from Windows.
